src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java

```diff
@@ -353,6 +366,56 @@ private static BigQueryTableFieldSchema generateTableFieldSchema(Schema.Field fi
     return fieldSchema;
   }
 
+  public static com.google.cloud.bigquery.Schema convertCdapSchemaToBigQuerySchema(Schema schema) {
+    List<Schema.Field> inputFields = Objects.requireNonNull(schema.getFields(), "Schema must have fields");
+    List<com.google.cloud.bigquery.Field> fields = inputFields.stream()
+      .map(BigQuerySinkUtils::convertCdapFieldToBigQueryField)
+      .collect(Collectors.toList());
+    return com.google.cloud.bigquery.Schema.of(fields);
+  }
+
+  private static Field convertCdapFieldToBigQueryField(Schema.Field field) {
```

Comment:
@sechegaray can you walk through this code and let me know if I did this correctly? I'm essentially mapping CDAP types into LegacySQLTypes, and adding scale/precision for numeric fields.

Note that this also does a recursive call to handle nested records.